tic_tac_toe.js

```diff
@@ -0,0 +1,92 @@
+$(document).ready(function() {
+    function Cell(el) {
+        this.move = null;
+        this.el = el;
+        this.clicked = false;
+    }
+    Cell.prototype.handleCellClick = function(rowNum, colNum, board) {
+        if (!board.table[rowNum][colNum].clicked && !board.gameOver) {
+            board.table[rowNum][colNum].clicked = true;
+            board.table[rowNum][colNum].move = board.turn;
+            this.el.html(board.turn);
+            var turn = "O";
+            if (board.turn === turn) { turn = "X"; }
+            board.turn = turn;
+            board.totalMoves++;
+            if (board.totalMoves > 4) {
+                var winner = board.checkForWinner(board);
+                if (winner) {
+                    console.log(winner);
+                    board.gameOver = true;
+                }
+            }
+        }
+    };
+    function Board() {
+        this.table = [];
+        this.turn = "X";
+        this.totalMoves = 0;
+        this.gameOver = false;
+    }
+    Board.prototype.horizontalWin = function(i, self) {
+        var move = board.table[i][0].move;
+        if (move === board.table[i][1].move &&
+                move === board.table[i][2].move) {
+            return move;
+        }
+        return null;
+    };
+    Board.prototype.verticalWin = function(self) {
+        for (var i = 0; i < self.table.length; i++) {
+            var move = self.table[0][i].move;
+            if (move === self.table[1][i].move &&
+                    move === self.table[2][i].move) {
+                return move;
+            }
+        }
+        return null;
+    };
+    Board.prototype.diagonalWin = function(i, self) {
+        var move = self.table[0][i].move;
+        if (move === self.table[1][1].move &&
+                move === self.table[2][2-i].move) {
+            return move;
+        }
+        return null;
+    };
```

Comment:
These win scenarios all seem pretty similar, can you think of a way to DRY this up a tad?